gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -226,8 +230,21 @@ public void sendOutstanding() {
       currentOpenBatch = new Batch<>(prototype, batchingDescriptor, batchingSettings, batcherStats);
     }
 
-    final ApiFuture<ResponseT> batchResponse =
-        unaryCallable.futureCall(accumulatedBatch.builder.build());
+    final ApiFuture<ResponseT> batchResponse;
+    if (unaryCallable instanceof TracedBatchedContextCallable) {
```

Comment:
Why do we need this instanceof? Instanceofs are areally not something we would want to use unless absolutely necessary, since it is usually an indication of mistakes in OOP design of the application.

Reply:
This is a workaround.

The general idea here is that we want to export state out a batcher so that we can send it to opencensus and eventually to dataflow counters. We dont want to couple the implementation of the tracing to the batcher. So we want to pass the state as a call context. The only 2 approaches I can think of are:
1. try to insert it in the existing ApiCallContext and have the next callable probe for it
2. have a special callable and have Batcher probe it

The second approach ended up creating a lot less breakage and least amount of overhead